internal/entry.go

```diff
@@ -18,6 +18,19 @@ type entry struct {
 }
 
 func NewEntry(ctx Context, time time.Time) Entry {
+	return newEntry(ctx, time)
+}
+
+func NewEntryFromString(ctx Context, stringTime string) (Entry, error) {
```

Comment:
Since this function is aimed at editing entries (existing or new), I would probably name the function appropriately. 
```
func EditEntry(ctx Context, time string) (Entry, error) {
}
```